ModulBot/Controllers/MessageController.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System;
+using System.Threading.Tasks;
+using Telegram.Bot.Types;
+using Microsoft.AspNetCore.Mvc;
+using Telegram.Bot.Types.Enums;
+using ModulBot.Models;
+using System.Linq;
+using ModulBot.Commands;
+using ModulBot.Cards;
+namespace ModulBot.Controllers
+{
+    [Route("api/message/update")]
+    public class MessageController : Controller
+    {
+        [HttpPost]
+        public async Task<OkResult> Post([FromBody]Update update)
+        {
+            var botClient = await Bot.GetBotClientAsync();
+
+            var user = DataModel.RememberUser(update.Message);
+
+            if (update == null) return Ok();
+
+
+            try
+            {
+                if (update.CallbackQuery != null)
+                {
+                    var message = update.CallbackQuery.Message;
+                    ButtonAction(update.CallbackQuery.Data, message.Chat.Id, message.MessageId);
+                }
+
+                else if (update.Message != null && update.Message.Type == MessageType.Text)
+                {
+                    var message = update.Message;
+
+                    var str = message.Text.Trim();
+                    if (str == "/createstat" || Bot.IsDialogStart1)
+                    {
+                        CreateStat stat = new CreateStat();
+                        stat.Run(message);
+                        return Ok();
+                    }
+                    var indexOfSpace = str.IndexOf(" ", StringComparison.Ordinal);
+                    var command = indexOfSpace != -1 ? str.Substring(0, indexOfSpace).ToLower() : str.ToLower();
+                    if (command[0] != '/')
+                    {
+                        command = "/" + command;
+                    }
+
+                    var help = new Help();
+
+                    var tool = Bot.Tools.FirstOrDefault(x => x.CommandsName.Any(y => y.Equals(command)));
+                    if (tool != null)
+                    {
+                        message.Text = indexOfSpace >= 0 ? message.Text.Substring(indexOfSpace, str.Length - indexOfSpace) : String.Empty;
+                        //message.Text = indexOfSpace >= 0 ? message.Text.Substring(indexOfSpace + 1, str.Length - indexOfSpace - 1) : String.Empty;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this commented out code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://172.21.13.172:9091/coding_rules#rule_key=csharpsquid%3AS125)